### ISSUE TITLE
Build errors with Xcode 9 beta 5

### ISSUE THREAD

**Issue by ldiqual** [nvzqz/RandomKit#42]
On 5.2.2 (via cocoapods):

```
INFO [2017-08-08 11:13:36.61]: ▸ Signing /Users/distiller/Library/Developer/Xcode/DerivedData/Scoop-apzwymdrenvrifcyulsguqmakzdj/Build/Products/Debug-iphonesimulator/RxSwift/RxSwift.framework
INFO [2017-08-08 11:13:36.61]: ▸ Building Pods/RandomKit [Debug]
INFO [2017-08-08 11:13:36.61]: ▸ Check Dependencies
INFO [2017-08-08 11:13:36.62]: ▸ Compiling ARC4Random.swift
INFO [2017-08-08 11:13:36.63]: ▸ ❌  /Users/distiller/scoop-ios/Pods/RandomKit/Sources/RandomKit/Types/RandomGenerator/ARC4Random.swift:67:20: argument labels '(extendingOrTruncating:)' do not match any available overloads
INFO [2017-08-08 11:13:36.63]: ▸ return UInt16(extendingOrTruncating: random32())
INFO [2017-08-08 11:13:36.63]: ▸                    ~~~~~~~~~^~~~~~~~~~~~
INFO [2017-08-08 11:13:36.68]: ▸ ❌  /Users/distiller/scoop-ios/Pods/RandomKit/Sources/RandomKit/Types/RandomGenerator/ARC4Random.swift:76:20: argument labels '(extendingOrTruncating:)' do not match any available overloads
INFO [2017-08-08 11:13:36.68]: ▸ return UInt8(extendingOrTruncating: random32())
INFO [2017-08-08 11:13:36.68]: ▸                    ^
INFO [2017-08-08 11:13:36.77]: ▸ Compiling ChaCha.swift
INFO [2017-08-08 11:13:36.78]: ▸ ❌  /Users/distiller/scoop-ios/Pods/RandomKit/Sources/RandomKit/Types/RandomGenerator/ChaCha.swift:134:25: argument labels '(extendingOrTruncating:)' do not match any available overloads
INFO [2017-08-08 11:13:36.78]: ▸ _state.12 = UInt32(extendingOrTruncating: low)
INFO [2017-08-08 11:13:36.78]: ▸                    ^
INFO [2017-08-08 11:13:36.84]: ▸ ❌  /Users/distiller/scoop-ios/Pods/RandomKit/Sources/RandomKit/Types/RandomGenerator/ChaCha.swift:135:25: argument labels '(extendingOrTruncating:)' do not match any available overloads
INFO [2017-08-08 11:13:36.84]: ▸ _state.13 = UInt32(extendingOrTruncating: low &>> 32)
INFO [2017-08-08 11:13:36.84]: ▸                         ^
INFO [2017-08-08 11:13:36.91]: ▸ ❌  /Users/distiller/scoop-ios/Pods/RandomKit/Sources/RandomKit/Types/RandomGenerator/ChaCha.swift:136:25: argument labels '(extendingOrTruncating:)' do not match any available overloads
INFO [2017-08-08 11:13:36.91]: ▸ _state.14 = UInt32(extendingOrTruncating: high)
INFO [2017-08-08 11:13:36.91]: ▸                         ^
INFO [2017-08-08 11:13:36.95]: ▸ 2017-08-08 11:13:36.955 xcodebuild[1480:17340] Error Domain=IDETestOperationsObserverErrorDomain Code=14 "Test operation was canceled. If you believe this error represents a bug, please attach the log file at /var/folders/42/bp51_0vn1kv6lqhv0wmfjrqw0000gn/T/com.apple.dt.XCTest/IDETestRunSession-4BA227F8-1C2D-4764-96CF-6535D7354DF3/ScoopTests-2CB9ACA0-44E2-4AA0-9C74-3BC465ABC5A9/Session-ScoopTests-2017-08-08_111333-MOGaxv.log" UserInfo={NSLocalizedDescription=Test operation was canceled. If you believe this error represents a bug, please attach the log file at /var/folders/42/bp51_0vn1kv6lqhv0wmfjrqw0000gn/T/com.apple.dt.XCTest/IDETestRunSession-4BA227F8-1C2D-4764-96CF-6535D7354DF3/ScoopTests-2CB9ACA0-44E2-4AA0-9C74-3BC465ABC5A9/Session-ScoopTests-2017-08-08_111333-MOGaxv.log}
INFO [2017-08-08 11:13:36.95]: ▸ Testing failed:
INFO [2017-08-08 11:13:36.95]: ▸ Argument labels '(extendingOrTruncating:)' do not match any available overloads
INFO [2017-08-08 11:13:36.95]: ▸ Argument labels '(extendingOrTruncating:)' do not match any available overloads
INFO [2017-08-08 11:13:36.95]: ▸ Argument labels '(extendingOrTruncating:)' do not match any available overloads
INFO [2017-08-08 11:13:36.95]: ▸ Argument labels '(extendingOrTruncating:)' do not match any available overloads
INFO [2017-08-08 11:13:36.95]: ▸ Argument labels '(extendingOrTruncating:)' do not match any available overloads
INFO [2017-08-08 11:13:36.95]: ▸ Argument labels '(extendingOrTruncating:)' do not match any available overloads
INFO [2017-08-08 11:13:36.95]: ▸ Non-nominal type 'Self' does not support explicit initialization
INFO [2017-08-08 11:13:36.95]: ▸ Binary operator '|' cannot be applied to operands of type 'Self' and 'Int'
INFO [2017-08-08 11:13:36.95]: ▸ Argument labels '(extendingOrTruncating:)' do not match any available overloads
INFO [2017-08-08 11:13:36.95]: ▸ Argument labels '(extendingOrTruncating:)' do not match any available overloads
INFO [2017-08-08 11:13:36.96]: ▸ Argument labels '(extendingOrTruncating:)' do not match any available overloads
INFO [2017-08-08 11:13:36.96]: ▸ Argument labels '(extendingOrTruncating:)' do not match any available overloads
INFO [2017-08-08 11:13:36.96]: ▸ Argument labels '(extendingOrTruncating:)' do not match any available overloads
INFO [2017-08-08 11:13:36.96]: ▸ Expression type 'Bool' is ambiguous without more context
INFO [2017-08-08 11:13:36.96]: ▸ ** TEST FAILED **
INFO [2017-08-08 11:13:36.96]: ▸ The following build commands failed:
INFO [2017-08-08 11:13:36.96]: ▸ CompileSwiftSources normal i386 com.apple.xcode.tools.swift.compiler
INFO [2017-08-08 11:13:36.96]: ▸ CompileSwift normal i386 /Users/distiller/scoop-ios/Pods/RandomKit/Sources/RandomKit/Types/RandomGenerator/ARC4Random.swift
INFO [2017-08-08 11:13:36.96]: ▸ CompileSwift normal i386 /Users/distiller/scoop-ios/Pods/RandomKit/Sources/RandomKit/Types/RandomGenerator/ChaCha.swift
INFO [2017-08-08 11:13:36.96]: ▸ CompileSwift normal i386 /Users/distiller/scoop-ios/Pods/RandomKit/Sources/RandomKit/Extensions/Swift/Integer+RandomKit.swift
INFO [2017-08-08 11:13:36.96]: ▸ CompileSwift normal i386 /Users/distiller/scoop-ios/Pods/RandomKit/Sources/RandomKit/Types/RandomGenerator/RandomBytesGenerator.swift
INFO [2017-08-08 11:13:36.96]: ▸ (5 failures)
INFO [2017-08-08 11:13:36.97]: ▸ ❌  /Users/distiller/scoop-ios/Pods/RandomKit/Sources/RandomKit/Types/RandomGenerator/ChaCha.swift:137:25: argument labels '(extendingOrTruncating:)' do not match any available overloads
INFO [2017-08-08 11:13:36.97]: ▸ _state.15 = UInt32(extendingOrTruncating: high &>> 32)
INFO [2017-08-08 11:13:36.97]: ▸                         ^
INFO [2017-08-08 11:13:37.02]: ▸ Compiling Integer+RandomKit.swift
INFO [2017-08-08 11:13:37.02]: ▸ ❌  /Users/distiller/scoop-ios/Pods/RandomKit/Sources/RandomKit/Extensions/Swift/Integer+RandomKit.swift:580:20: non-nominal type 'Self' does not support explicit initialization
INFO [2017-08-08 11:13:37.03]: ▸ let bits = Self(extendingOrTruncating: MemoryLayout<Self>.size * 8 - 1)
INFO [2017-08-08 11:13:37.03]: ▸                         ^
INFO [2017-08-08 11:13:37.03]: ▸ ❌  /Users/distiller/scoop-ios/Pods/RandomKit/Sources/RandomKit/Extensions/Swift/Integer+RandomKit.swift:359:69: binary operator '|' cannot be applied to operands of type 'Self' and 'Int'
INFO [2017-08-08 11:13:37.03]: ▸ return random(withMaxWidth: width, using: &randomGenerator) | (1 &<< Self(width - 1))
INFO [2017-08-08 11:13:37.03]: ▸                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
INFO [2017-08-08 11:13:37.03]: ▸ Compiling RandomBytesGenerator.swift
INFO [2017-08-08 11:13:37.03]: ▸ ❌  /Users/distiller/scoop-ios/Pods/RandomKit/Sources/RandomKit/Types/RandomGenerator/RandomBytesGenerator.swift:49:20: argument labels '(extendingOrTruncating:)' do not match any available overloads
INFO [2017-08-08 11:13:37.03]: ▸ return UInt32(extendingOrTruncating: randomBytes())
INFO [2017-08-08 11:13:37.03]: ▸                                                                     ^
INFO [2017-08-08 11:13:37.08]: ▸ ❌  /Users/distiller/scoop-ios/Pods/RandomKit/Sources/RandomKit/Types/RandomGenerator/RandomBytesGenerator.swift:58:20: argument labels '(extendingOrTruncating:)' do not match any available overloads
INFO [2017-08-08 11:13:37.08]: ▸ return UInt16(extendingOrTruncating: randomBytes())
INFO [2017-08-08 11:13:37.08]: ▸                    ^
INFO [2017-08-08 11:13:37.14]: ▸ ❌  /Users/distiller/scoop-ios/Pods/RandomKit/Sources/RandomKit/Types/RandomGenerator/RandomBytesGenerator.swift:67:20: argument labels '(extendingOrTruncating:)' do not match any available overloads
INFO [2017-08-08 11:13:37.14]: ▸ return UInt8(extendingOrTruncating: randomBytes())
INFO [2017-08-08 11:13:37.14]: ▸                    ^
INFO [2017-08-08 11:13:37.22]: ▸ ❌  /Users/distiller/scoop-ios/Pods/RandomKit/Sources/RandomKit/Types/RandomGenerator/RandomBytesGenerator.swift:90:20: argument labels '(extendingOrTruncating:)' do not match any available overloads
INFO [2017-08-08 11:13:37.23]: ▸ return UInt16(extendingOrTruncating: randomBytes())
INFO [2017-08-08 11:13:37.23]: ▸                    ^
INFO [2017-08-08 11:13:37.28]: ▸ ❌  /Users/distiller/scoop-ios/Pods/RandomKit/Sources/RandomKit/Types/RandomGenerator/RandomBytesGenerator.swift:99:20: argument labels '(extendingOrTruncating:)' do not match any available overloads
INFO [2017-08-08 11:13:37.28]: ▸ return UInt8(extendingOrTruncating: randomBytes())
INFO [2017-08-08 11:13:37.28]: ▸    
```

**Comment by nvzqz** [comment]
Released [v5.2.3](https://github.com/nvzqz/RandomKit/releases/tag/v5.2.3) with this fixed.

**Comment by ldiqual** [comment]
@nvzqz thanks a lot for the quick turnaround!
